hi-ml-histopathology/src/histopathology/models/deepmil.py

```diff
@@ -352,20 +352,95 @@ def get_bag_label(labels: Tensor) -> Tensor:
         # SlidesDataModule attributes a single label to a bag of tiles already no need to do majority voting
         return labels
 
+    @staticmethod
+    def get_empty_lists(shape: int, n: int) -> List:
+        ll = []
+        for _ in range(n):
+            ll.append([None] * shape)
+        return ll
+
+    @staticmethod
+    def get_patch_coordinate(slide_offset: List, patch_location: List, patch_size: List) -> Tuple[int, int, int, int]:
```

Comment:
can you maybe precise the type of the lists?  is it List[int]?

Reply:
Also it would be nice if you could add doc strings to say what to expect for each argument, as you did for patch_location?

Reply:
Added the type, but I see below a lot of suggestions about moving this to use the Box class. I'll change